core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -96,35 +147,73 @@ public GeoBoundingBoxQueryBuilder bottomRight(String geohash) {
      * @param lon The longitude
      */
     public GeoBoundingBoxQueryBuilder bottomLeft(double lat, double lon) {
-        box[BOTTOM] = lat;
-        box[LEFT] = lon;
+        if (topLeft != null) {
+            topLeft.resetLon(lon);
+        } else {
+            topLeft = new GeoPoint(Double.NaN, lon);
+        }
+
+        if (bottomRight != null) {
+            bottomRight.resetLon(lon);
```

Comment:
I always screw one of these up once the knot in my brain is large enough. Thanks for catching this.